### PR TITLE
[GLUTEN-2627][VL] support map extract

### PR DESCRIPTION
**Pull request by summaryzb**
## What changes were proposed in this pull request?
(Fixes: [ISSUE-2627](https://github.com/oap-project/gluten/issues/2627))
fixes: https://github.com/oap-project/gluten/issues/2587

## How was this patch tested?
manual tests

### CODE REVIEW

**Review by jackylee-ch — DISMISSED** [review]

**Comment by jackylee-ch** [comment]
cc @zhouyuan @zhztheplayer

File: gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala [hunk 1]
```diff
@@ -368,12 +368,21 @@ object ExpressionConverter extends SQLConfHelper with Logging {
           c.children.map(child => replaceWithExpressionTransformer(child, attributeSeq))
         new CreateMapTransformer(substraitExprName.get, children, c.useStringTypeWhenEmpty, c)
       case g: GetMapValue =>
-        new GetMapValueTransformer(
-          substraitExprName.get,
-          replaceWithExpressionTransformer(g.child, attributeSeq),
-          replaceWithExpressionTransformer(g.key, attributeSeq),
-          g.failOnError,
-          g)
+        if (BackendsApiManager.isVeloxBackend) {
```
Comment (zhouyuan) [inline gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala:371]: nit:  move these logic into velox/ck backendapi?

Comment (jackylee-ch) [reply gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala:371]: @summaryzb A simple example `genAliasTransformer`.

Comment (summaryzb) [reply gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala]: make sense

**Review by jackylee-ch — APPROVED** [review]

**Pull request merged**